src/features/contracts/contracts-directory.ts

```diff
@@ -8,13 +8,82 @@ import * as stream from 'stream';
 import * as path from 'path';
 import * as os from 'os';
 import type { RepositoryInfo, Contract } from './index';
+import { isValidUrl } from '../../lib/utils';
 
 const pipeline = util.promisify(stream.pipeline);
 const exists = util.promisify(fs.exists);
 const mkdir = util.promisify(fs.mkdir);
 
 const CONTRACTS_BASE_DIR = path.join(os.tmpdir(), 'contracts');
 
+const getDataUri = async (
+	filePath: string,
+	mimeType: 'image/png' | 'image/svg+xml',
+) => {
+	const base64Content = (await fs.promises.readFile(filePath)).toString(
```

Comment:
Are logotypes stored on disk?

Reply:
@dfunckt They are included as part of the contracts repo, and when we sync the contracts we basically pull the entire repo tarball, so yes, they will be on disk together with the contract json

Reply:
Hmm, I didn't realise that we were doing that for contracts -- we should either move away from this sort of writing into the container or discuss exposing a mount point where storage can be bound.

Reply:
@dfunckt why do you think we shouldn't be writing in the container? We only write to `tmp` storage and we don't really care if we lose that data as we will refetch it on the next sync.

Reply:
Containers should generally be "immutable" @sradevski -- by silently writing into the container you're making the assumption that there's arbitrarily large underlying storage available, which is not always the case and is what bind mounts and volumes are for (which make the storage requirement explicit and lets one plan accordingly).

Reply:
I don't think that's necessarily true, since the `contracts` data is essentially a cache, I don't see a difference between storing that in memory or on disk inside the container. In this case, I explicitly don't want the contract data to be persisted outside of a container's lifecycle, which is why I didn't want to use volumes. It doesn't really matter much if we use volumes TBH, I just don't see how that would be a better option here.

Reply:
I guess we could use `tmpfs` (?) which seems to be the docker way for all the use cases that `/tmp` offers.
See: https://docs.docker.com/storage/tmpfs/

Reply:
@thgreasi I was also thinking about that, but this is only read once every 5-10 minutes and I don't think it's worth keeping it in memory (since tmpfs doesn't write anything on disk). Anyhow, we should settle on one approach and we can go with that, but I personally find just storing it on disk to be the best option for the contracts.